### PR TITLE
CLOUD-777: support adding k9 policy statements into existing bucket policy

### PR DESCRIPTION
**Pull request by skuenzli**
This change introduces support for adding k9 policy statements into an existing existing bucket policy.

This change is necessary in order to support AWS CDK S3 Bucket features including autoDeleteObjects and inventory report delivery.  Issue #12 describes how autoDeleteObject's behavior prompted this change.

### CODE REVIEW

File: bin/k9-cdk.ts [hunk 1]
```diff
@@ -53,6 +54,35 @@ const k9BucketPolicyProps: k9.s3.K9BucketPolicyProps = {
...
+            allowPrincipalArns: readConfigArns,
+        },
+        {
+            accessCapability: k9.k9policy.AccessCapability.WriteData,
+            allowPrincipalArns: writeDataArns,
+        }
+    )
+};
+
+k9.s3.makeBucketPolicy(stack, 'AutoDeleteBucket', k9AutoDeleteBucketPolicyProps);
```
Comment (skuenzli) [inline bin/k9-cdk.ts:82]: @brentryan how does this usage look to you?

What do you think `k9.s3.makeBucketPolicy` should be named?

I'm leaning towards `k9.s3.grantAccess`

Comment (brentryan) [reply bin/k9-cdk.ts:82]: `k9.s3.grantAccess` follows the conventions that CDK is using. The only thing that might be slightly different is most of "grantXXX()" methods are for applying policy changes to IAM. When it's a resource policy, it's less consistent I've found. It's usually something like bucket.addToResourcePolicy() ...

I think my personal preference is for the "grantXXX" style.

Comment (skuenzli) [reply bin/k9-cdk.ts:82]: Agree - I'll improve function names and merge this PR today.

**Comment by skuenzli** [comment]
Updated grantAccessViaResourcePolicy to collect and return a list of `aws-iam.AddToResourcePolicyResult`

This approach:
* converges closer to how a call to `S3Bucket.addToResourcePolicy` works
* provides information others might need to understand dependencies (`AddToResourcePolicyResult` has an `IDependable` member)
* (possibly) eliminates confusion as to whether the Policy is on the bucket (it is)

Most people will probably ignore all this and say:
```
k9.s3.grantAccessViaResourcePolicy(stack, "S3Bucket", k9BucketPolicyProps);
```

which is basically the same as before.

KMS key policy definition will be provided similar semantics in an upcoming change.

**Pull request merged**